mitodl/mit-open
# Topics Listing Page

### What are the relevant tickets?
- Closes https://github.com/mitodl/hq/issues/4186

### Description (What does it do?)
This PR:
- Adds the topics listing page
- Updates our `<Chip />` component to align better (both styles and terminology) with https://www.figma.com/design/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=104-1187&m=dev

### Screenshots (if appropriate):

<img width="1593" alt="Screenshot 2024-05-23 at 1 34 15 PM" src="https://github.com/mitodl/mit-open/assets/9010790/ec5c3bc6-f244-48c9-83fa-a371e9b65b72">

<img width="1611" alt="Screenshot 2024-05-23 at 1 09 55 PM" src="https://github.com/mitodl/mit-open/assets/9010790/086218b3-4693-4a15-b695-9fe39a72f85a">

<img width="646" alt="Screenshot 2024-05-23 at 1 10 16 PM" src="https://github.com/mitodl/mit-open/assets/9010790/ca611558-7b70-4fac-b1e0-830b49278554">



### How can this be tested?
1. View http://localhost:8063/topics/. You should:
    - see root-level topics, along with child "subtopics", displayed as above.
    - Clicking a topic or subtopic should take you to the corresponding channel page
2. Check that you can navigate to the topics page via:
    - the "By topics" link in upper-left nav drawer (click "Explore MIT")
    - the "See All" link on homepage
3. I also made some changes to our "Chip" component to better align with the [designs](https://www.figma.com/design/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=104-1187&m=dev). Review these:
    - Visit the storybook pages for Chips http://localhost:6006/?path=/docs/smoot-design-chip-pill--docs ... They should look like designs.
        - If you're running in docker, storybook should start automatically. It crashes sometines, and can be restarted with `yarn storybook`. 
    - View the homepage. There are chips right below the searchbar they should look good.
    - The date chip on course cards should still look similar to before (though course cards are not finalized).
- Check the departments page still looks good. I updated it to use the same Banner component for header that topics page uses. The Figma designs for the two pages use the same component, so they should update in sync.

### Additional Context
